shared/middleware/helpers/source_bucket.rb

```diff
@@ -66,6 +66,63 @@ def restore_previous_version(encrypted_channel_id, filename, version_id, user_id
     response.to_h
   end
 
+  # Copies the files in the src_channel to the dest_channel
+  # Update the animation manifest to include the version ids of the animations
+  # in dest_channel
+  # Note: this function assumes that the animations have already been copied
+  def remix_source(src_channel, dest_channel, options={})
```

Comment:
Cleaning up this function probably starts with tearing out unused features.  For example:
- In the only place you call this method, you don't pass an `options` argument, so let's tear out that argument and everything dependent on it.
- We never use the return value of this method, so you can tear out `result`.
- I see you've tested the case where we put something other than a well-formatted main.json in the sources bucket - that's great.  We might make some stronger assumptions about main.json though, or make sure to skip a bunch of logic if we're not going to need it.